src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -443,6 +443,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - <none>
     void ControlCore::_sendInputToConnection(std::wstring_view wstr)
     {
+        if (wstr.empty())
+        {
+            return;
+        }
+
+        // The connection may call functions like WriteFile() which may block indefinitely.
+        // It's important we don't hold any mutexes across such calls.
+        _terminal->_assertUnlocked();

```

Comment:
@lhecker This assertion fails (and crashes Terminal if built in Debug mode without debugger) when restoring pane which exited with a non-zero code. Is this expected?

https://github.com/microsoft/terminal/assets/12915102/d6280b8e-c005-4269-9d83-715f2cf3f052

Reply:
No it's not expected. Thank you for the video! I'll try to fix this issue soon.

In the meantime, please feel free to simply comment out the assertion. 🙂

Reply:
#17266